src/src/PluginStructs/P143_data_struct.cpp

```diff
@@ -428,12 +425,12 @@ bool P143_data_struct::plugin_write(struct EventStruct *event,
         # if P143_FEATURE_INCLUDE_M5STACK
         case P143_DeviceType_e::M5StackEncoder:
         {
-          if (_useOffset) { // Adjust offset
+          if (_useOffset) {                          // Adjust offset
             int16_t encoderCount = I2C_readS16_LE_reg(_i2cAddress, P143_M5STACK_REG_ENCODER);
             _offsetEncoder = encoderCount - _encoderPosition;
-          } else {          // Set position using upgraded firmware
-            int16_t encoderCount = _encoderPosition;
-            I2C_write16_reg(_i2cAddress, P143_M5STACK_REG_ENCODER, BYTESWAP16(encoderCount));
+          } else {                                   // Set position using upgraded firmware
+            int16_t encoderCount = _encoderPosition; // Forced int16_t to uint16_t conversion is intentional
```

Comment:
This comment is confusing.
Why allocate an int first and then directly use it in `byteSwap16`?
After it is not used anyway, so there doesn't seem to be a need to allocate a new variable.

Reply:
The conversion to the `int16_t` is used to keep the sign, then the bytes are swapped, is the intention. I'll re-check if that works as intended...

Reply:
Should better have used the `I2C_write16_LE_reg()` function (missed that earlier, though it's where that byte-swap originated...), but now disabled (using a `#define`) all M5Stack encoder firmware V1.1 related code, as we don't have that device available (yet).